addons/marketing_campaign/models/marketing_campaign.py

```diff
@@ -0,0 +1,133 @@
+                                      help='If set, this field will help segments that work in "no duplicates" mode to avoid '
+                                             'selecting similar records twice. Similar records are records that have the same value for '
+                                             'this unique field. For example by choosing the "email_from" field for CRM Leads you would prevent '
+                                             'sending the same campaign to the same email address again. If not set, the "no duplicates" segments '
+                                             "will only avoid selecting the same record again if it entered the campaign previously. "
+                                             "Only easily comparable fields like textfields, integers, selections or single relationships may be used.")
+
+    mode = fields.Selection([('test', 'Test Directly'),
+                         ('test_realtime', 'Test in Realtime'),
+                         ('manual', 'With Manual Confirmation'),
+                         ('active', 'Normal')], required=True, default='test', help="""Test - It creates and process all the activities directly (without waiting for the delay on transitions) but does not send emails or produce reports.
+Test in Realtime - It creates and processes all the activities directly but does not send emails or produce reports.
+With Manual Confirmation - the campaigns runs normally, but the user has to validate all workitem manually.
+Normal - the campaign runs normally and automatically sends all emails and reports (be very careful with this mode, you're live!)""")
+    state = fields.Selection([('draft', 'New'),
+                          ('running', 'Running'),
+                          ('cancelled', 'Cancelled'),
+                          ('done', 'Done')],
+                         string='Status', copy=False, default='draft')
+    activity_ids = fields.One2many('marketing.campaign.activity',
+                                   inverse_name='campaign_id', string='Activities')
+    fixed_cost = fields.Float(
+        help="Fixed cost for running this campaign. You may also specify variable cost and revenue on each campaign activity. Cost and Revenue statistics are included in Campaign Reporting.", digits=dp.get_precision('Product Price'))
+    segment_ids = fields.One2many(
+        'marketing.campaign.segment', inverse_name='campaign_id', string='Segments', readonly=False)
+    segments_count = fields.Integer(
+        compute='_count_segments', string='Segments')
+
+    @api.multi
+    def _count_segments(self):
+            Counts_data = self.env['marketing.campaign.segment'].read_group([('campaign_id', 'in', self.ids)], ['campaign_id'], ['campaign_id'])
+            mapped_data = dict([(m['campaign_id'][0], m['campaign_id_count']) for m in Counts_data])
+            for segments in self:
```

Comment:
self is a campaign object. so use "campaign" variable name instead of "segments". it's misleading.